XenModel/PoolJoinRules.cs

```diff
@@ -673,5 +687,55 @@ public static bool HasCompatibleManagementInterface(IXenConnection slaveConnecti
             else
                 return slaveConnection.Cache.PIFs.Any(p => !p.physical && p.management && p.VLAN != -1);
         }
+
+        public static bool HasIpForClusterNetwork(IXenConnection masterConnection, Host slaveHost, out bool clusterHostInBond)
+        {
+
+            var clusterHost = masterConnection.Cache.Cluster_hosts.FirstOrDefault();
+
+            if (clusterHost == null)
+            {
+                clusterHostInBond = false;
+                return true;
+            }
+            var clusterHostPif = clusterHost.Connection.Resolve(clusterHost.PIF);
+            clusterHostInBond = clusterHostPif.IsBondNIC();

```

Comment:
`Resolve` can return null, so we should check clusterHostPif for null before using it.